js/exportsms.js

```diff
@@ -417,7 +411,116 @@ var exportsms = function() {
    *   Unfinished config object we're in the process of building. 
    * @return Story object
    */
-  function _buildEndGame(endGameData, config) {
+  function _buildEndGame(endLevelData, endGameData, config) {
+    var endGameObject,
+        endGameDatum,
+        endLevelDatum,
+        i,j,k,l,m,n,p,q,
+        min, 
+        max,
+        superlativeArray,
+        superlative,
+        rank,
+        oip,
+        choiceObject
+        ;
+
+    endGameObject = {
+      "__comments": "",
+      "indiv-message-end-game-format": "",
+      "group-message-end-game-format": "",
+      "choices": [],
+      "group-level-success-oips": [],
+      "group-success-failure-oips": {
+        "0": null,
+        "1": null,
+        "2": null,
+        "3": null,
+        "4": null,
+        "5": null,
+        "6": null
+      },
+      "indiv-level-success-oips": [],
+      "indiv-rank-oips": {
+        "1": null,
+        "2": null,
+        "3": null,
+        "4": null,
+        "1-tied": null,
+        "2-tied": null
+      }
+    };
+
+    // Iterating through all the endgame data.
+    for (j = 0; j < endGameData.length; j++) {
+      endGameDatum = endGameData[j];
+      // Storing oip for an individual rank result.
+      if (endGameDatum.type === 'passageEndGameIndivRankResult') {
```

Comment:
I'd suggest using `PassageEndGameIndivRankResult.prototype.defaults.type` instead of the hardcoded string.

Same deal below with `'passageEndGameIndivSuperlativeResult'`, `'passageEndGameGroupSuccessNumberResult'` and `'passageEndLevelGroup'`.